src/all/comicgrowl/src/eu/kanade/tachiyomi/extension/all/comicgrowl/ComicGrowl.kt

```diff
@@ -1,63 +1,330 @@
+                        jsonObject["imageUrl"]!!.jsonPrimitive.content.toHttpUrl().newBuilder().fragment(scramble)
+                            .build()
+                    pageList.add(
+                        Page(
+                            index = jsonObject["sort"]!!.jsonPrimitive.int,
+                            imageUrl = imageUrl.toString(),
+                        ),
+                    )
+                }
+            }
+        }
+        return pageList
+    }
+
+    override fun imageRequest(page: Page): Request {
+        val newHeaders = headersBuilder()
+            .set("Referer", baseUrl)
+            .build()
+        return GET(page.imageUrl!!, newHeaders)
+    }
+
+    private fun imageDescrambler(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val rawRequest = request.newBuilder().url(request.url.newBuilder().fragment(null).build()).build()
+        val response = chain.proceed(rawRequest)
+
+        if (request.url.fragment.isNullOrEmpty()) {
+            return response
+        }
+
+        val scramble = request.url.fragment!!
+        val tiles = buildList {
+            scramble.split("-").forEachIndexed { index, s ->
+                val scrambleInt = s.toInt()
+                add(index, TilePos(scrambleInt / 4, scrambleInt.mod(4)))
+            }
+        }
+
+        val scrambledImg = BitmapFactory.decodeStream(response.body.byteStream())
+        val descrambledImg = drawDescrambledImage(scrambledImg, scrambledImg.width, scrambledImg.height, tiles)
+
+        val output = ByteArrayOutputStream()
+        descrambledImg.compress(Bitmap.CompressFormat.JPEG, 90, output)
+
+        val image = output.toByteArray()
+        val body = image.toResponseBody("image/jpeg".toMediaType())
+
+        return response.newBuilder()
+            .body(body)
+            .build()
+    }
+
+    private fun drawDescrambledImage(rawImage: Bitmap, width: Int, height: Int, tiles: List<TilePos>): Bitmap {
+        // Prepare canvas
+        val descrambledImg = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888)
+        val canvas = Canvas(descrambledImg)
+
+        // Tile width and height(4x4)
+        val tileWidth = width / 4
+        val tileHeight = height / 4
+
+        // Draw rect
+        var count = 0
+        for (x in 0..3) {
+            for (y in 0..3) {
+                val desRect = Rect(x * tileWidth, y * tileHeight, (x + 1) * tileWidth, (y + 1) * tileHeight)
+                val srcRect = Rect(
+                    tiles[count].x * tileWidth,
+                    tiles[count].y * tileHeight,
+                    (tiles[count].x + 1) * tileWidth,
+                    (tiles[count].y + 1) * tileHeight,
+                )
+                canvas.drawBitmap(rawImage, srcRect, desRect, null)
+                count++
+            }
+        }
+        return descrambledImg
+    }
+
+    private data class TilePos(val x: Int, val y: Int) // Left-top corner position
```

Comment:
Don't need a data class for this.
```suggestion
    private class TilePos(val x: Int, val y: Int) // Left-top corner position
```